README.md

```diff
@@ -143,96 +312,56 @@ Available types, formats and resultant value of the cast:
 <sup>3</sup> default format returns String  
 <sup>4</sup> topojson is not implemented
 
-### Infer
+### Resource
 
-Given headers and data, `infer` will return a JSON Table Schema as a JSON object based on the data values. Given the data file, example.csv:
+A javascript model of a resource (schema+source of data)
 
-```csv
-id,age,name
-1,39,Paul
-2,23,Jimmy
-3,36,Jane
-4,28,Judy
-```
+Instance always returns `Promise`. In case if schema object is not valid, it will reject promise.
 
-Call `infer` with headers and values from the datafile:
+Source of data can be:
+* array of rows with values
+* local CSV file
+* remote CSV file (URL)
+* readable stream
 
-```javascript
-import parse from 'csv-parse'
-import fs from 'fs'
-import { infer } from 'jsontableschema'
-
-fs.readFile('example.csv', (err, data) => {
-  parse(data, (error, values) => {
-    const headers = values.shift()
-        , schema = infer(headers, values)
-  })
-})
-```
+Following methods are available on `Resource` instances:
+* `iter(callback, failFast, skipConstraints)`<sup>1,2</sup> - iterate through the given dataset provided in constructor and returns converted data
```

Comment:
Resource class currently has very thin API, and no plans yet to add something new. But I agree, that principe YAGNI here will be good :)

skipConstraints if set to true just skip checking constraints, which might be set for the field.